### PR TITLE
Bug 1856399 - Use distinct version code in Fenix AAB

### PR DESCRIPTION
**Pull request by gbrownmozilla**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1856399

### CODE REVIEW

**Comment by gbrownmozilla** [comment]
Clear the 'g' bit in Fenix version codes used in AAB builds, to ensure that Fenix AAB version codes are always distinct from Fenix APK version codes.

I feel like there should be a more elegant way to detect AAB-ness without passing -Paab ... suggestions?

There are similar issues for Focus and Reference-Browser. Once this is sorted out, I can handle those cases in separate PRs.

**Comment by jcristau** [comment]
I'm wondering if it'd be better to switch it around so the aab is the one with the higher version code?
Or does it not really matter because once we start shipping the bundle we'll stop uploading apks?

**Comment by gbrownmozilla** [comment]
Right, I think we would never push APKs and AAB for the same build to Google Play...but who knows what surprises the future holds? 

I've flipped around the 'g' bit to be 1 for AAB, and also set the 'p' and 'x' bits for AAB: That should guarantee that the AAB version code is > any APK version code created at the same time.

File: android-components/plugins/config/src/main/java/ConfigPlugin.kt [hunk 1]
```diff
@@ -144,17 +144,19 @@ object Config {
...
 
         // 'p' bit is 1 for 64-bit architectures.
-        if (abi == "arm64-v8a" || abi == "x86_64") {
+        if (aab || abi == "arm64-v8a" || abi == "x86_64") {
             version = version or (1 shl 1)
         }
 
-        // 'g' bit is currently always 1 (see comment above)
-        version = version or (1 shl 0)
+        // 'g' bit is 0 for APK, 1 for AAB
```
Comment (JohanLorenzo) [inline android-components/plugins/config/src/main/java/ConfigPlugin.kt:156]: 👍

**Review by JohanLorenzo — APPROVED** [review]
Changes make sense to me! I'm glad we're able to reemploy the smallest digit here. Thanks for updating the comments 👍

**Pull request merged**